player/src/main/java/com/focusit/jsflight/player/webdriver/SeleniumDriver.java

```diff
@@ -56,25 +68,49 @@ public SeleniumDriver(UserScenario scenario, Map<String, String> externalLastUrl
     {
         this.scenario = scenario;
         this.lastUrls = externalLastUrls;
+        //This must be set due to equals of WebElement
+        NO_OP_ELEMENT.setId("NO_OP");
+    }
+
+    public void setUseRandomChars(boolean useRandomChars)
+    {
+        this.useRandomChars = useRandomChars;
+    }
+
+    public void setPageTimeoutMs(int pageTimeoutMs)
+    {
+        this.pageTimeoutMs = pageTimeoutMs;
+    }
+
+    public void setCheckPageJs(String checkPageJs)
+    {
+        this.checkPageJs = checkPageJs;
+    }
+
+    public void setLookupScript(String lookupScript)
+    {
+        this.lookupScript = lookupScript;
+    }
+
+    public void setUiShownScript(String uiShownScript)
+    {
+        this.uiShownScript = uiShownScript;
+    }
+
+    public void setMaxElementGroovy(String maxElementGroovy)
+    {
+        this.maxElementGroovy = maxElementGroovy;
     }
 
     public void closeWebDrivers()
     {
         drivers.values().forEach(WebDriver::close);
     }
 
-    public WebElement findTargetWebElement(String script, WebDriver wd, JSONObject event, String target)
+    public WebElement findTargetWebElement(WebDriver wd, JSONObject event, String target)
     {
-        try
-        {
-            LOG.info("looking for " + target);
-            return wd.findElement(By.xpath(target));
-        }
-        catch (NoSuchElementException e)
-        {
-            LOG.info("failed looking for {}. trying to restore xpath");
-            return (WebElement)new PlayerScriptProcessor(scenario).executeWebLookupScript(script, wd, target, event);
-        }
+        waitPageReady(wd, event);
+        return (WebElement)new PlayerScriptProcessor(scenario).executeWebLookupScript(lookupScript, wd, target, event);
```

Comment:
You may wrap a WebElement into a LocatedElement just here, aren't you? What was the reason to make a very special wrapper for WebDriver to force every component use LocatedElement?